### PR TITLE
Collection Data

### PR DESCRIPTION
**Pull request by maheshsattala**
Added CRUD operations for the CollectionData and added unit and integration tests for the same.

## Summary by CodeRabbit

- **New Features**
	- Introduced new API endpoints for managing collection data: save, update, delete, and list.
	- Enhanced data validation for collection data, ensuring integrity and error handling.

- **Bug Fixes**
	- Improved error handling for various invalid conditions when saving and updating collection data.

- **Tests**
	- Added comprehensive tests for the new collection data API endpoints, covering edge cases for saving, updating, deleting, and retrieving collection data.
	- Expanded unit tests for the `CognitionDataProcessor` class to ensure robust data management functionality.

### CODE REVIEW

File: kairon/shared/cognition/processor.py [hunk 1]
```diff
@@ -112,6 +112,112 @@ def list_cognition_schema(self, bot: Text):
...
+            if not data or not isinstance(data, dict):
+                raise AppException("Invalid value for data")
+
+    def save_collection_data(self, payload: Dict, user: Text, bot: Text):
+        collection_name = payload.get("collection_name", None)
+        data = payload.get('data')
+        is_secure = payload.get('is_secure')
+        CognitionDataProcessor.validate_collection_payload(collection_name, is_secure, data)
+
+        if Utility.is_exist(CollectionData, bot=bot, collection_name__iexact=collection_name,
```
Comment (hiteshghuge) [inline kairon/shared/cognition/processor.py:140]: collection name can be duplicate, as we will maintaing data against single type(collection_name)

File: kairon/shared/cognition/processor.py [hunk 2]
```diff
@@ -112,6 +112,112 @@ def list_cognition_schema(self, bot: Text):
...
+    def update_collection_data(self, collection_id: str, payload: Dict, user: Text, bot: Text):
+        from kairon import Utility
+
+        collection_name = payload.get("collection_name", None)
+        data = payload.get('data')
+        is_secure = payload.get('is_secure')
+        CognitionDataProcessor.validate_collection_payload(collection_name, is_secure, data)
+
+        if not Utility.is_exist(CollectionData, bot=bot, collection_name__iexact=collection_name,
+                                raise_error=False):
```
Comment (hiteshghuge) [inline kairon/shared/cognition/processor.py:164]: same as above

**Review by hiteshghuge — APPROVED** [review]
reviewed
